src/message_iter.c

```diff
@@ -326,6 +343,25 @@ static int ldbus_message_iter_close_container(lua_State *L) {
 	return 1;
 }
 
+static int ldbus_message_iter_gc(lua_State *L)
+{
+	lDBusMessageIter *iter = luaL_checkudata(L, -1, DBUS_MESSAGE_ITER_METATABLE);
+
+	if (iter->message) {
+		dbus_message_unref(iter->message);
+		iter->message = NULL;
+	}
```

Comment:
```suggestion
	unref_ldbus_message_iter(iter);
```

(and you'll need to reorder the functions)